zerver/views/streams.py

```diff
@@ -518,27 +532,86 @@ def remove_subscriptions_backend(
     return json_success(request, data=result)
 
 
-def you_were_just_subscribed_message(
-    acting_user: UserProfile, recipient_user: UserProfile, stream_names: Set[str]
+def make_message_for_subscription_change(
+    acting_user: UserProfile,
+    recipient_user: UserProfile,
+    stream_names: Set[str],
+    *,
+    add_subscription: bool = False,
+    remove_subscription: bool = False,
 ) -> str:
     subscriptions = sorted(stream_names)
-    if len(subscriptions) == 1:
-        with override_language(recipient_user.default_language):
+    message: str = ""
+    with override_language(recipient_user.default_language):
+        if add_subscription and len(subscriptions) == 1:
             return _("{user_full_name} subscribed you to the stream {stream_name}.").format(
                 user_full_name=silent_mention_syntax_for_user(acting_user),
                 stream_name=f"#**{subscriptions[0]}**",
             )
-
-    with override_language(recipient_user.default_language):
-        message = _("{user_full_name} subscribed you to the following streams:").format(
-            user_full_name=silent_mention_syntax_for_user(acting_user),
-        )
+        elif remove_subscription and len(subscriptions) == 1:
+            return _("{user_full_name} unsubscribed you from the stream {stream_name}.").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+                stream_name=f"#**{subscriptions[0]}**",
+            )
+        elif add_subscription:
+            message = _("{user_full_name} subscribed you to the following streams:").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+            )
+        else:
+            message = _("{user_full_name} unsubscribed you from the following streams:").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+            )
     message += "\n\n"
     for stream_name in subscriptions:
         message += f"* #**{stream_name}**\n"
     return message
 
 
+def send_messages_for_removed_subscribers(
+    removed_subscribers: List[Tuple[UserProfile, Stream]], acting_user: UserProfile
+) -> None:
+    if len(removed_subscribers) == 0:
+        return
+
+    user_id_to_unsubscribed_streams: Dict[Any, Set[str]] = {}
+    user_id_to_user: Dict[Any, UserProfile] = {}
+    for user, stream in removed_subscribers:
+        if user.id in user_id_to_unsubscribed_streams:
+            user_id_to_unsubscribed_streams[user.id].add(stream.name)
+        else:
+            user_id_to_unsubscribed_streams[user.id] = {stream.name}
+            user_id_to_user[user.id] = user
+
+    mention_backend = MentionBackend(acting_user.realm_id)
+    all_messages = []
```

Comment:
Let's call it `message_requests` -- this is how we name messages which are yet to be sent.

Reply:
Renamed variable.